File: java/socketd/src/main/java/org/noear/socketd/transport/core/stream/StreamAcceptorRequest.java
package org.noear.socketd.transport.core.stream;

import org.noear.socketd.transport.core.*;

import java.util.concurrent.CompletableFuture;

/**
 * 请求流接收器
 *
 * @author noear
 * @since 2.0
 */
public class StreamAcceptorRequest extends StreamAcceptorBase {
    private final CompletableFuture<Reply> future;

    public StreamAcceptorRequest(String sid, long timeout, CompletableFuture<Reply> future) {
        super(sid, timeout);
        this.future = future;
    }

    /**
     * 是否单发接收
     */
    @Override
    public boolean isSingle() {
        return true;
    }

    /**
     * 是否结束接收
     */
    @Override
    public boolean isDone() {
        return future.isDone();
    }

    /**
     * 接收时
     */
    @Override
    public void onAccept(MessageInternal message, Channel channel) {
        future.complete(message);
    }
}

File: python/socketd/socketd/transport/core/StreamAcceptorRequest.py

from socketd.transport.core.CompletableFuture import CompletableFuture
from socketd.transport.core.StreamAcceptorBase import StreamAcceptorBase


class StreamAcceptorRequest(StreamAcceptorBase):
    def __init__(self, future, timeout):
        self.future: CompletableFuture = future
        self.timeout = timeout

    def is_single(self):
        return True

    def is_done(self):
        return self.future.done()

    def timeout(self):
        return self.timeout

    def accept(self, message, onError):
        self.future.set_result(message)